upload_eia_data/contents/src/update_eia_data_on_carto.py

```diff
@@ -70,47 +72,46 @@ def upload_to_aws(local_file, bucket, s3_file):
         logging.error("Credentials not available")
         return False
 
-def fetch_eia_data(table_name):
+def fetch_eia_data(table_name, country_table):
     '''
     This function fetches and processes data from the EIA API for a specified Carto table
     INPUT   table_name: name of Carto table we want to fetch and process data for (string)
+            country_table: name of countries and regions we want to fetch and process data for (dataframe)
     RETURN  all_eia_data: dataframe of processed data for this table (pandas dataframe)
     '''
-    # pull the eia catogory id that is included in this table
-    category_id = eia_rw_table.loc[table_name, 'eia_category_id']
-    eia_unit = eia_rw_table.loc[table_name, 'eia_unit']
-
-    # insert the url used to find the series ids of the data 
-    url = 'https://api.eia.gov/category/?api_key={}&category_id={}'.format(EIA_KEY, category_id)
-
-    # fetch the information of all series of data in this category from the API 
-    r = requests.get(url)
-    series = r.json()['category']['childseries']
-
-    # store the names, ids and units of the child series in a list of dictionaries
-    ids = [{'country': ', '.join(child['name'].split(', ')[1:-1]), 'series_id': child['series_id'], 'units': child['units']} for child in series]
-    # subset the dictionary list by EIA unit
-    ids = [id for id in ids if id['units'] == eia_unit]
+    # pull the eia activityId, productId, unit that is included in this table info
+    activityId = eia_rw_table.loc[table_name, 'activityId']
+    productId = eia_rw_table.loc[table_name, 'productId']
+    unit = eia_rw_table.loc[table_name, 'unit']
 
     # create an empty dataframe to store data 
     df = pd.DataFrame()
-    # loop through each series id
-    for id in ids:
-        # construct the API call to fetch data from the series
-        data_url = 'https://api.eia.gov/series/?api_key={}&series_id={}'.format(EIA_KEY, id['series_id'])
-        logging.info('Fetching data for {}'.format(id['country']))
+    # loop through each country/region
+    for alpha_3_code in country_table['Alpha-3 code']:
+        # construct the API call to fetch data from the country
+        url = 'https://api.eia.gov/v2/international/data/?api_key={}&frequency=annual&data[0]=value&facets[activityId][]={}&facets[productId][]={}&facets[countryRegionId][]={}&facets[unit][]={}'.format(EIA_KEY, activityId, productId, alpha_3_code, unit)
```

Comment:
@weiqi-tori how do you feel about standardizing on f-strings for improved readability?

Reply:
I guess there are 6 places in this file that would need to be updated.

Reply:
Updated